component/api/src/main/java/org/exoplatform/portal/mop/service/LayoutService.java

```diff
@@ -311,4 +324,35 @@ default void saveImportStatus(Status status) { // NOSONAR
     throw new UnsupportedOperationException();
   }
 
+  /**
+   * Retrieves sites according to the given site filter
+   * 
+   * @param siteFilter {@link SiteFilter} site filter for results filtering
+   * @return {@link List} of sites portal config
+   */
+  default List<PortalConfig> getSites(SiteFilter siteFilter) {
+    throw new UnsupportedOperationException();
+  }
+
+  /**
+   * Retrieve site banner as stream
+   *
+   * @param siteName name of the portal site
+   * @return found {@link InputStream}
+   * @throws ObjectNotFoundException When site not found or file attachment
+   */
+  default InputStream getSiteBannerStream(String siteName) throws ObjectNotFoundException, IOException {
+    throw new UnsupportedOperationException();
+  }
+
+  /**
+   * remove site banner
+   *
+   * @param siteName name of the portal site
+   * @throws ObjectNotFoundException When site not found or file attachment
+   */
+  default void removeSiteBanner(String siteName) throws ObjectNotFoundException {
+    throw new UnsupportedOperationException();
+  }
```

Comment:
Just a though, we have an Image Portlet, shouldn't we use it instead ? @rdenarie @margondicco

Reply:
@boubaker the image portlet allows to upload an image for each instance, in our case the banner is specific to the site entity. Besides, we should display the site banner into the second level component of hamburger menu portlet and not into a portlet.

Reply:
ok, I see. I just made a though for it to reuse the same strategy as new Image Portlet, but it's ok for me.